api/notd/collection_activity_processor.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+
+from core.store.retriever import DateFieldFilter
+from core.store.retriever import Direction
+from core.store.retriever import Order
+from core.store.retriever import StringFieldFilter
+from core.util import date_util
+
+from notd.model import RetrievedCollectionHourlyActivity
+from notd.store.retriever import Retriever
+from notd.store.schema import BlocksTable
+from notd.store.schema import TokenTransfersTable
+
+
+class CollectionActivityProcessor:
+
+    def __init__(self,retriever: Retriever) -> None:
+        self.retriever = retriever
+
+    async def calculate_collection_hourly_activity(self, registryAddress: str, date: datetime.datetime) -> RetrievedCollectionHourlyActivity:
+        tokenTransfers = await self.retriever.list_token_transfers(
+            fieldFilters=[
+                StringFieldFilter(TokenTransfersTable.c.registryAddress.key, eq=registryAddress),
+                DateFieldFilter(BlocksTable.c.blockDate.key, gte=date_util.datetime_from_datetime(dt=date, hours=-1)),
+                DateFieldFilter(BlocksTable.c.blockDate.key, lt=date),
+            ],
+            orders=[Order(fieldName=TokenTransfersTable.c.value.key, direction=Direction.DESCENDING)],
+        )
+        saleCount = transferCount = totalVolume = averageValue = minimumValue = 0
+        if len(tokenTransfers) > 0:
+            for tokenTransfer in tokenTransfers:
+                if tokenTransfer.value != 0:
+                    minimumValue = min(float('inf'),tokenTransfer.value)
+                    saleCount += 1
+                transferCount += tokenTransfer.amount
+                totalVolume += tokenTransfer.value
+                averageValue += totalVolume/transferCount
+
+            maximumValue = tokenTransfers[0].value
+        else:
+            maximumValue = 0
```

Comment:
why is this here?

Reply:
in fact, where is minimumValue and saleCount even defined if there are no sales?

Reply:
So I should makes those Optional yh? and just have them as null?

Reply:
then why is just maximumValue here?